File: api/src/main/java/net/impactdev/gts/api/blacklist/Blacklist.java
package net.impactdev.gts.api.blacklist;

import com.google.common.collect.Multimap;
import net.kyori.adventure.key.Key;

public interface Blacklist {

	Multimap<Class<?>, String> getBlacklist();

	void append(Class<?> registrar, String key);

	boolean isBlacklisted(Class<?> registrar, String query);

    void clear();
}

File: api/src/main/kotlin/net/impactdev/gts/api/blacklist/Blacklist.kt
package net.impactdev.gts.api.blacklist

import com.google.common.collect.Multimap

interface Blacklist {
    val blacklist: Multimap<Class<*>?, String?>?
    fun append(registrar: Class<*>?, key: String?)
    fun isBlacklisted(registrar: Class<*>?, query: String?): Boolean
    fun clear()
}